// maiku.net/components/ui/admin/education-list.jsx
import React, { Component } from "react";
import PropTypes from "prop-types";

import School from "maiku.net/components/ui/admin/school";

class EducationList extends Component {
	render() {
		const { data, moveItem, remove, update } = this.props;

		return (
			<ul>
				<li>
					<div className="row" style={{ padding: "8px", fontWeight: "bold" }}>
						<div className="col-md-3">School</div>
					</div>
				</li>
				{data.map((item, index) => (
					<School
						key={item._id}
						index={index}
						item={item}
						moveItem={moveItem}
						remove={remove}
						update={update}
					/>
				))}
			</ul>
		);
	}
}

EducationList.propTypes = {
	data: PropTypes.arrayOf(
		PropTypes.shape({
			_id: PropTypes.string.isRequired,
			completed: PropTypes.bool.isRequired,
			description: PropTypes.string.isRequired,
			end: PropTypes.string,
			program: PropTypes.string.isRequired,
			school: PropTypes.string.isRequired,
			start: PropTypes.string.isRequired,
		}),
	),
	remove: PropTypes.func.isRequired,
	update: PropTypes.func.isRequired,
	moveItem: PropTypes.func.isRequired,
};

EducationList.defaultProps = {
	data: [],
};

export default EducationList;
