// maiku.net/components/ui/admin/education.jsx
import React, { Component } from "react";

import EducationList from "maiku.net/components/ui/admin/education-list"

export default class Education extends Component {
    state = {
        isCompleted: false
    }

    toggleCompleted = () => {
        this.setState((prevState, props) => { prevState.isCompleted = !prevState.isCompleted })
    }

    addEducation = event => {
        event.preventDefault();

        const uuidV4 = require('uuid/v4');
        const education = {
            _id: uuidV4(),
            completed: this.state.isCompleted,
            school: this.school.value,
            start: this.start.value,
            end: this.end.value,
            program: this.program.value,
            description: this.description.value
        }

        this.props.addItem(education);

        this.form.reset();
    }

    render(){
        const { data } = this.props;
 
        return(
            <div>
                <h5 className="font-accident-two-normal uppercase subtitle">Manage Education</h5>

                <div className="col-md-12">
                    <div className="dividewhite2"></div>
                    <div className="responsive-table">
                        <EducationList
                            data={data}
                            removeItem={this.props.removeItem}
                            updateItem={this.props.updateItem}
                        />
                    </div>
                </div>

                <div className="col-md-12">
                    <div className="dividewhite2"></div>
                    <div className="responsive-table">
                        <form
                            className="wpcf7-form"
                            onSubmit={(event) => this.addEducation(event)}
                            ref={(input) => { this.form = input }}
                        >
                            <div className="field">
                                <input
                                    className="form-control"
                                    type="text"
                                    id="school"
                                    name="school"
                                    placeholder="School"
                                    required
                                    ref={(input) => { this.school = input }}
                                />
                            </div>

                            <div className="form-inline">
                                <div className="form-group" style={{ paddingRight: "10px" }}>
                                    <input
                                        className="form-control"
                                        type="text"
                                        id="start"
                                        name="start"
                                        placeholder="Start"
                                        required
                                        ref={(input) => { this.start = input }}
                                    />
                                </div>
                                <div className="form-group">
                                    <input
                                        className="form-control"
                                        type="text"
                                        id="end"
                                        name="end"
                                        placeholder="End"
                                        ref={(input) => { this.end = input }}
                                    />
                                </div>

                                <div className="form-group checkbox radio-inline">
                                    <input 
                                        className="form-control"
                                        type="checkbox"
                                        id="completed"
                                        name="completed"
                                        onChange={this.toggleCompleted}
                                    />

                                    <label>Completed? </label>
                                </div>
                            </div>

                            <div className="field">
                                <input
                                    className="form-control"
                                    type="text"
                                    id="program"
                                    name="program"
                                    placeholder="Program"
                                    required
                                    ref={(input) => { this.program = input }}
                                />
                            </div>

                            <div className="field">
                                <textarea
                                    className="form-control"
                                    type="text"
                                    id="description"
                                    name="description"
                                    placeholder="Description"
                                    required
                                    ref={(input) => { this.description = input }}
                                >
                                </textarea>
                            </div>

                            <div className="field text-center">
                                <button type="submit" className="btn btn-lg btn-darker">Add</button>
                            </div>
                        </form>
                    </div>
                </div>
            </div>
        );
    }
}
