// maiku.net/components/ui/admin/employment.jsx
import React, { PureComponent } from "react";

import EmployerList from "maiku.net/components/ui/admin/employer-list";

export default class Employment extends PureComponent {

    addEmployer = event => {
            event.preventDefault();
            const uuidV4 = require('uuid/v4');
            this.props.addItem({
                _id: uuidV4(),
                current: this.end.value === "",
                company: this.company.value,
                start: this.start.value,
                end: this.end.value,
                position: this.position.value,
                description: this.description.value
            });

            this.form.reset();
    }

    render() {
        const { data, removeItem, updateItem, addProject, removeProject, updateProject } = this.props;

        return (
            <div>
                <h5 className="font-accident-two-normal uppercase subtitle">Manage Employment</h5>

                <div className="col-md-12">
                    <div className="dividewhite2"></div>
                    <div className="responsive-table">
                        <EmployerList
                            data={data}
                            removeItem={removeItem}
                            updateItem={updateItem}
                            addProject={addProject}
                            removeProject={removeProject}
                            updateProject={updateProject}
                        />
                    </div>
                </div>

                <div className="col-md-12">
                    <div className="dividewhite2"></div>
                    <div className="responsive-table">
                        <form
                            className="wpcf7-form"
                            onSubmit={(event) => this.addEmployer(event)}
                            ref={(input) => { this.form = input }}
                        >
                            <div className="field">
                                <input
                                    className="form-control"
                                    type="text"
                                    id="company"
                                    name="company"
                                    placeholder="Employer"
                                    required
                                    ref={(input) => { this.company = input }}
                                />
                            </div>

                            <div className="form-inline">
                                <div className="form-group" style={{ paddingRight: "10px" }}>
                                    <input
                                        className="form-control"
                                        type="text"
                                        id="start"
                                        name="start"
                                        placeholder="Start"
                                        required
                                        ref={(input) => { this.start = input }}
                                    />
                                </div>
                                <div className="form-group">
                                    <input
                                        className="form-control"
                                        type="text"
                                        id="end"
                                        name="end"
                                        placeholder="End"
                                        ref={(input) => { this.end = input }}
                                    />
                                </div>
                            </div>

                            <div className="field">
                                <input
                                    className="form-control"
                                    type="text"
                                    id="position"
                                    name="position"
                                    placeholder="Position"
                                    required
                                    ref={(input) => { this.position = input }}
                                />
                            </div>

                            <div className="field">
                                <textarea
                                    className="form-control"
                                    type="text"
                                    id="description"
                                    name="description"
                                    placeholder="Description"
                                    required
                                    ref={(input) => { this.description = input }}
                                >
                                </textarea>
                            </div>

                            <div className="field text-center">
                                <button type="submit" className="btn btn-lg btn-darker">Add</button>
                            </div>
                        </form>
                    </div>
                </div>
            </div>
        );
    }
}
