// maiku.net/components/ui/admin/employer-list.jsx
import React, { Component } from "react";
import PropTypes from "prop-types";

import Job from "maiku.net/components/ui/admin/job";

export default class EmployerList extends Component {
	render() {
		const { data, moveItem, remove, update } = this.props;

		return (
			<ul>
				<li>
					<div className="row" style={{ padding: "8px", fontWeight: "bold" }}>
						<div className="col-md-3">Employer</div>
					</div>
				</li>
				{data.map((item, index) => (
					<Job
						key={item._id}
						index={index}
						item={item}
						moveItem={moveItem}
						remove={remove}
						update={update}
					/>
				))}
			</ul>
		);
	}
}

EmployerList.propTypes = {
	data: PropTypes.arrayOf(
		PropTypes.shape({
			_id: PropTypes.string.isRequired,
			company: PropTypes.string.isRequired,
			current: PropTypes.bool.isRequired,
			description: PropTypes.string.isRequired,
			end: PropTypes.string,
			position: PropTypes.string.isRequired,
			start: PropTypes.string.isRequired,
		}),
	).isRequired,
	remove: PropTypes.func.isRequired,
	update: PropTypes.func.isRequired,
	moveItem: PropTypes.func.isRequired,
};

EmployerList.defaultProps = {
	data: [],
};
