// maiku.net/components/ui/admin/fact-list.jsx
import React, { Component } from "react";
import PropTypes from "prop-types";
import { DragDropContext } from "react-dnd";
import HTML5Backend from "react-dnd-html5-backend";

import Fact from "maiku.net/components/ui/admin/fact";

class FactList extends Component {
	render() {
		const { data, moveItem, remove, update } = this.props;

		return (
			<ul>
				<li>
					<div className="row" style={{ padding: "8px", fontWeight: "bold" }}>
						<div className="col-md-3">Label</div>
						<div className="col-md-6">Fact</div>
					</div>
				</li>
				{data.map((item, index) => (
					<Fact key={item._id} index={index} item={item} moveItem={moveItem} remove={remove} update={update} />
				))}
			</ul>
		);
	}
}

FactList.propTypes = {
	data: PropTypes.arrayOf(
		PropTypes.shape({
			_id: PropTypes.string.isRequired,
			fact: PropTypes.string.isRequired,
			label: PropTypes.string.isRequired,
		}),
	).isRequired,
	moveItem: PropTypes.func.isRequired,
	remove: PropTypes.func.isRequired,
	update: PropTypes.func.isRequired,
};

FactList.defaultProps = {
	data: [],
};

export default DragDropContext(HTML5Backend)(FactList);
