// maiku.net/components/ui/admin/facts.jsx
import React, { Component } from "react";
import PropTypes from "prop-types";

import FactList from "maiku.net/components/ui/admin/facts-list";

export default class Facts extends Component {

    addItem = event => {
        event.preventDefault();

        const uuidV4 = require('uuid/v4');
        this.props.addItem({
            _id: uuidV4(),
            label: this.label.value,
            fact: this.fact.value
        });

        this.form.reset();
    }
    render(){
        const { data, removeItem, updateItem } = this.props;

        return(
            <div>
                 <h5 className="font-accident-two-normal uppercase subtitle">Facts About Me</h5>

                 <div className="col-md-12">
                    <div className="dividewhite2"></div>
                    <div className="responsive-table">
                        <FactList
                            data={data}
                            removeItem={removeItem}
                            updateItem={updateItem}
                        />
                    </div>
                </div>

                <div className="col-md-12">
                    <div className="dividewhite2"></div>
                    <div id="form-messages"></div>

                    <form
                        className="wpcf7-form"
                        onSubmit={(event) => this.addItem(event)}
                        ref={(input) => { this.form = input }}
                    >
                        <div className="field">
                            <input
                                className="form-control"
                                type="text"
                                id="label"
                                name="label"
                                placeholder="Label"
                                required
                                ref={(input) => { this.label = input }}
                            />
                        </div>

                        <div className="field">
                            <input
                                className="form-control"
                                type="text"
                                id="fact"
                                name="fact"
                                placeholder="Fact"
                                required
                                ref={(input) => { this.fact = input }}
                            />
                        </div>

                        <div className="field text-center">
                            <button type="submit" className="btn btn-lg btn-darker">Add</button>
                        </div>

                    </form>

                </div>
            </div>
        );
    }
}

Facts.propTypes = { 
    data: PropTypes.arrayOf(PropTypes.shape({
        _id: PropTypes.string.isRequired,
        fact: PropTypes.string.isRequired,
        label: PropTypes.string.isRequired,
    })).isRequired,
    addItem: PropTypes.func.isRequired,
    removeItem: PropTypes.func.isRequired,
    updateItem: PropTypes.func.isRequired,
}

Facts.defaultProps = {
    data: [],
}
