// maiku.net/components/ui/admin/info.jsx
import React, { PureComponent } from "react";
import PropTypes from "prop-types";

import InfoList from "maiku.net/components/ui/admin/info-list"

export default class Info extends PureComponent {
    state = {
        data: this.props.data
    };

    handleChange(event) {
        this.setState({ data: event.target.value });
        console.log(event.target.value)
    }

    componentWillReceiveProps(nextProps) {
        this.setState({ data: nextProps.data })
    }

    addInfo = event => {
        event.preventDefault();
        const uuidV4 = require('uuid/v4');

        this.props.addItem({
            _id: uuidV4(),
            value: this.info.value
        })

        this.form.reset();
    }

    updateInfo = event => {
        event.preventDefault();
    }

    render() {
        const numberOfInfoItems = !this.state.data ? 0 : this.state.data.length;
        const { removeItem, updateItem } = this.props;
        return (
            <div>
                <h5 className="font-accident-two-normal uppercase subtitle">Information list</h5>
            
                <div className="col-md-12">
                    <div className="dividewhite2"></div>
                    <div className="responsive-table">
                        <InfoList
                            info={this.props.data} 
                            removeItem={removeItem}
                            updateItem={updateItem}
                            />
                    </div>
                </div>

                <div className="col-md-12">
                    <div className="dividewhite2"></div>
                    <div id="form-messages"></div>

                    <form
                        className="wpcf7-form"
                        onSubmit={(event) => this.addInfo(event)}
                        ref={(input) => { this.form = input }}
                    >
                        <div className="field">
                            <input
                                className="form-control"
                                type="text"
                                id="info"
                                name="info"
                                placeholder="Information"
                                required
                                ref={(input) => { this.info = input }}
                            />
                        </div>

                        <div className="field text-center">
                            {numberOfInfoItems < 4 && 
                            <button type="submit" className="btn btn-lg btn-darker">Add</button>
                            }
                        </div>

                    </form>

                </div>
            </div>
        );
    }
}

Info.propTypes = { 
    data: PropTypes.arrayOf(PropTypes.shape({
        _id: PropTypes.string,
        value: PropTypes.string,
    })).isRequired,
    addItem: PropTypes.func.isRequired,
    removeItem: PropTypes.func.isRequired,
    updateItem: PropTypes.func.isRequired,
}

Info.defaultProps = {
    data: [],
}
