// maiku.net/components/ui/admin/info-list.jsx
import React, { Component } from "react";
import PropTypes from "prop-types";

export default class InfoList extends Component {
	state = {
		editing: null,
	};

	toggleEditing = itemId => {
		this.setState(() => ({
			editing: itemId,
		}));
	};

	handleEditField = event => {
		if (event.keyCode === 13) {
			const target = event.target;
			const update = {};

			update._id = this.state.editing;
			update[target.name] = target.value;

			this.handleUpdate(update);
		}
	};

	handleEditItem = () => {
		const itemId = this.state.editing;

		this.handleUpdate({
			_id: itemId,
			value: this.value.value,
		});
	};

	handleUpdate = update => {
		this.props.update("info", update);
		this.setState(() => ({
            editing: null,
        }));
	};

	renderItemOrEditField = item => {
		const { remove } = this.props;

		if (this.state.editing === item._id) {
			return (
				<tr key={`editing-${item._id}`}>
					<td>
						<input
							onKeyDown={event => this.handleEditField(event)}
							type="text"
							className="form-control"
							ref={`value_${item._id}`}
							name="value"
							defaultValue={item.value}
						/>
					</td>
					<td>
						<button className="btn btn-default" onClick={() => this.handleEditItem()}>Update</button>
					</td>
				</tr>
			);
		}
		return (
			<tr key={item._id} onClick={() => this.toggleEditing(item._id)}>
				<td>{item.value}</td>
				<td>
					<button className="btn btn-default" onClick={() => remove("info", item)}>Remove</button>
				</td>
			</tr>
		);
	};

	render() {
		const { info } = this.props;

		return (
			<table className="table">
				<thead>
					<tr>
						<td>Information</td>
						<td className="width-15" />
					</tr>
				</thead>
				<tbody>
					{info.map(item => this.renderItemOrEditField(item))}
				</tbody>
			</table>
		);
	}
}

InfoList.propTypes = {
	info: PropTypes.arrayOf(
		PropTypes.shape({
			_id: PropTypes.string,
			value: PropTypes.string,
		}),
	).isRequired,
	remove: PropTypes.func.isRequired,
	update: PropTypes.func.isRequired,
};

InfoList.defaultProps = {
	info: [],
};
