// maiku.net/components/ui/admin/info-list.jsx
import React, { Component } from "react";
import PropTypes from "prop-types";
import { DragDropContext } from "react-dnd";
import HTML5Backend from "react-dnd-html5-backend";

import Information from "maiku.net/components/ui/admin/information";

class InfoList extends Component {
	render() {
		const { data, moveItem, remove, update } = this.props;

		return (
			<ul>
				<li>
					<div className="row" style={{ padding: "8px", fontWeight: "bold" }}>
						<div className="col-md-3">Information</div>
					</div>
				</li>
				{data.map((item, index) => (
					<Information
						key={item._id}
						index={index}
						item={item}
						moveItem={moveItem}
						remove={remove}
						update={update}
					/>
				))}
			</ul>
		);
	}
}

InfoList.propTypes = {
	data: PropTypes.arrayOf(
		PropTypes.shape({
			_id: PropTypes.string,
			value: PropTypes.string,
		}),
	).isRequired,
	moveItem: PropTypes.func.isRequired,
	remove: PropTypes.func.isRequired,
	update: PropTypes.func.isRequired,
};

InfoList.defaultProps = {
	data: [],
};

export default DragDropContext(HTML5Backend)(InfoList);
