// maiku.net/components/ui/admin/job.jsx
import React, { Component } from "react";
import PropTypes from "prop-types";

import { DragSource, DropTarget } from "react-dnd";

import { ItemTypes } from "maiku.net/components/ui/constants";

const jobSource = {
	beginDrag(props) {
		return {
			id: props.id,
			index: props.index,
		};
	},
	endDrag(props, monitor) {
		const dropIndex = monitor.getItem().index;
		const dragIndex = props.index;

		props.moveItem("employment", dragIndex, dropIndex);
	},
};

const jobTarget = {
	drop(props, monitor) {
		const dragIndex = monitor.getItem().index;
		const dropIndex = props.index;

		// Don't replace items with themselves
		if (dragIndex === dropIndex) {
			return;
		}

		monitor.getItem().index = dropIndex;
	},
};

class Job extends Component {
	state = {
		editing: null,
	};

	toggleEditing = itemId => {
		this.setState(() => ({
			editing: itemId,
		}));
	};

	handleEditField = event => {
		if (event.keyCode === 13) {
			const target = event.target;
			const update = {};

			update._id = this.state.editing;
			update[target.name] = target.value;

			this.handleUpdate(update);
		}
	};

	handleEditItem = () => {
		const itemId = this.state.editing;

		this.handleUpdate({
			_id: itemId,
			current: this.start.value === "",
			company: this.company.value,
			start: this.start.value,
			end: this.end.value,
			position: this.position.value,
			description: this.description.value,
		});
	};

	handleUpdate = update => {
		this.props.update("employment", update);
		this.setState(() => ({
			editing: null,
		}));
	};

	toggleAddProject = item => {
		console.log(item, "add project");
	};

	renderEdit(item) {
		return (
			<li key={item._id}>
				<div className="row" style={{ padding: "8px", borderTop: "1px solid rgb(221, 221, 221)" }}>
					<div className="col-md-9">
						<div className="field">
							<input
								onKeyDown={event => this.handleEditField(event)}
								type="text"
								className="form-control"
								ref={ref => {
									this.company = ref;
								}}
								name="company"
								defaultValue={item.company}
							/>
						</div>

						<div className="form-inline">
							<div className="form-group" style={{ paddingRight: "10px" }}>
								<input
									className="form-control"
									type="text"
									id="start"
									name="start"
									placeholder="Start"
									required
									ref={ref => {
										this.start = ref;
									}}
									defaultValue={item.start}
								/>
							</div>
							<div className="form-group">
								<input
									className="form-control"
									type="text"
									id="end"
									name="end"
									placeholder="End"
									ref={ref => {
										this.end = ref;
									}}
									defaultValue={item.end}
								/>
							</div>
						</div>

						<div className="field">
							<input
								onKeyDown={event => this.handleEditField(event)}
								type="text"
								className="form-control"
								ref={ref => {
									this.position = ref;
								}}
								name="position"
								defaultValue={item.position}
							/>
						</div>

						<div className="field">
							<textarea
								type="text"
								className="form-control"
								ref={ref => {
									this.description = ref;
								}}
								name="description"
								defaultValue={item.description}
							/>
						</div>
					</div>
					<div className="col-md-3" style={{ textAlign: "right" }}>
						<button className="btn btn-default" onClick={() => this.handleEditItem()}>Update</button>
						&nbsp;
						<button className="btn btn-default" onClick={() => this.toggleEditing(null, false)}>
							Cancel
						</button>
					</div>
				</div>
			</li>
		);
	}

	render() {
		const { item, remove, connectDragSource, connectDropTarget } = this.props;

		if (this.state.editing === item._id) {
			return this.renderEdit(item);
		}

		return connectDragSource(
			connectDropTarget(
				<li key={item._id} onClick={() => this.toggleEditing(item._id)}>
					<div className="row" style={{ padding: "8px", borderTop: "1px solid rgb(221, 221, 221)" }}>
						<div className="col-md-9">{item.company}</div>
						<div className="col-md-3" style={{ textAlign: "right" }}>
							<button className="btn btn-default" onClick={() => remove("employment", item)}>Remove</button>
						</div>
					</div>
				</li>,
			),
		);
	}
}

Job = DropTarget(ItemTypes.INFORMATION, jobTarget, connect => ({
	connectDropTarget: connect.dropTarget(),
}))(Job);
Job = DragSource(ItemTypes.INFORMATION, jobSource, (connect, monitor) => ({
	connectDragSource: connect.dragSource(),
	isDragging: monitor.isDragging(),
}))(Job);

Job.propTypes = {
	item: PropTypes.shape({
		_id: PropTypes.string.isRequired,
		company: PropTypes.string.isRequired,
		current: PropTypes.bool.isRequired,
		description: PropTypes.string.isRequired,
		end: PropTypes.string,
		position: PropTypes.string.isRequired,
		start: PropTypes.string.isRequired,
	}).isRequired,
	remove: PropTypes.func.isRequired,
	update: PropTypes.func.isRequired,
	connectDragSource: PropTypes.func,
	connectDropTarget: PropTypes.func,
};

Job.defaultProps = {
	connectDragSource: () => {},
	connectDropTarget: () => {},
};

export default Job;
