// maiku.net/components/ui/admin/skills.jsx
import React, { Component } from "react";
import PropTypes from "prop-types";

import AddSkill from "maiku.net/components/ui/add-skill";
import SkillList from "maiku.net/components/ui/admin/skill-list";

export default class Skills extends Component {
	removeSkill = skill => {
		this.props.removeSkill(skill);
	};

	render() {
		return (
			<div>
				<h5 className="font-accident-two-normal uppercase subtitle">Manage skillset</h5>

				<div className="col-md-12">
					<div className="dividewhite2" />
					<div className="responsive-table">
						<SkillList
							skills={this.props.skills}
							removeSkill={this.props.removeSkill}
							updateSkill={this.props.updateSkill}
						/>
					</div>
				</div>

				<div className="col-md-12">
					<div className="dividewhite2" />
					<div id="form-messages" />

					<AddSkill addSkill={this.props.addSkill} />
				</div>
			</div>
		);
	}
}

Skills.propTypes = {
	skills: PropTypes.arrayOf(
		PropTypes.shape({
			_id: PropTypes.string.isRequired,
			level: PropTypes.number.isRequired,
			name: PropTypes.string.isRequired,
			type: PropTypes.string.isRequired,
		}),
	).isRequired,
	addSkill: PropTypes.func.isRequired,
	removeSkill: PropTypes.func.isRequired,
	updateSkill: PropTypes.func.isRequired,
};

Skills.defaultProps = {
	skills: [],
};
