// maiku.net/components/ui/admin/skills.jsx
import React from "react";
import PropTypes from "prop-types";

import AddSkill from "maiku.net/components/ui/admin/add-skill";
import SkillList from "maiku.net/components/ui/admin/skill-list";

const Skills = ({ skills, add, remove, update }) => (
	<div>
		<h5 className="font-accident-two-normal uppercase subtitle">Manage skillset</h5>

		<div className="col-md-12">
			<div className="dividewhite2" />
			<div className="responsive-table">
				<SkillList
					skills={skills}
					remove={remove}
					update={update}
				/>
			</div>
		</div>

		<div className="col-md-12">
			<div className="dividewhite2" />
			<div id="form-messages" />

			<AddSkill add={add} />
		</div>
	</div>
);

Skills.propTypes = {
	skills: PropTypes.arrayOf(
		PropTypes.shape({
			_id: PropTypes.string.isRequired,
			level: PropTypes.number.isRequired,
			name: PropTypes.string.isRequired,
			type: PropTypes.string.isRequired,
		}),
	).isRequired,
	add: PropTypes.func.isRequired,
	// addSkill: PropTypes.func.isRequired,
	remove: PropTypes.func.isRequired,
	update: PropTypes.func.isRequired,
};

Skills.defaultProps = {
	skills: [],
};

export default Skills;
