// maiku.net/components/ui/add-skill.jsx
import React, { Component } from "react";

export default class AddSkill extends Component {

    addSkill = event => {
        event.preventDefault();

        const uuidV4 = require('uuid/v4');

        const skill = { 
            _id: uuidV4(),
            type: this.type.value,
            name: this.name.value,
            level: this.level.value
        }

        this.props.addSkill(skill);
        this.form.reset();
    }

    render() {

        return (
            <form
                className="wpcf7-form"
                onSubmit={(event) => this.addSkill(event)}
                ref={(input) => { this.form = input }}
            >
                <select 
                    className="form-control"
                    ref={(input) => { this.type = input}}>
                    <option value="language">Language</option>
                    <option value="framework">Framework</option>
                    <option value="software">Software</option>
                    <option value="os">Operating System</option>
                </select>

                <div className="field">
                    <input
                        className="form-control"
                        type="text"
                        id="name"
                        name="name"
                        placeholder="Skill"
                        required
                        ref={(input) => { this.name = input }}
                    />
                </div>

                <div className="field">
                    <input
                        className="form-control"
                        type="number"
                        id="level"
                        name="level"
                        placeholder="Skill Level"
                        required
                        ref={(input) => { this.level = input }}
                    />
                </div>

                <div className="dividewhite2"></div>

                <div className="field text-center">
                    <button type="submit" className="btn btn-lg btn-darker">Add Skill</button>
                </div>
            </form>
        );
    }
}
