// maiku.net/components/ui/page-title.jsx
import React from "react";
import PropTypes from "prop-types";

const PageTitle = ({ origin, duration, delay, distance, name, title, info }) => (
	<section id="page-title" className="inner-section ui-block-color07">
		<div className="padding-50">
			<h2
				className="font-accident-two-extralight color01 uppercase"
				data-animation-origin={origin}
				data-animation-duration={duration}
				data-animation-delay={delay}
				data-animation-distance={distance}
			>
				{name}
			</h2>
			<h4
				className="font-accident-two-normal color01 uppercase subtitle"
				data-animation-origin={origin}
				data-animation-duration={duration}
				data-animation-delay={delay}
				data-animation-distance={distance}
			>
				{title}
			</h4>
			<p
				className="small color01"
				data-animation-origin={origin}
				data-animation-duration={duration}
				data-animation-delay={delay}
				data-animation-distance={distance}
			>
				{info}
			</p>
		</div>
	</section>
);

PageTitle.propTypes = {
	origin: PropTypes.string,
	duration: PropTypes.number,
	delay: PropTypes.number,
	distance: PropTypes.string,
	name: PropTypes.string.isRequired,
	title: PropTypes.string,
	info: PropTypes.string,
};

PageTitle.defaultProps = {
	origin: "right",
	duration: 400,
	delay: 100,
	distance: "50px",
	name: "Page Name",
	title: "",
	info: "",
};

export default PageTitle;
