// maiku.net/components/ui/resume-job.jsx
import React from "react";
import PropTypes from "prop-types";

const ResumeJob = ({origin, duration, delay, distance, start, end, company, position, description}) => (
    <li>
        <div className="timeline-badge primary"><i className="flaticon-identification28"></i></div>
        <div className="timeline-panel"
            data-animation-origin={origin}
            data-animation-duration={duration}
            data-animation-delay={delay}
            data-animation-distance={distance}>
            <p className="timeline-time fontcolor-invert">{start} - {end}</p>
            <div className="timeline-photo timeline-mark-01"></div>
            <div className="timeline-heading">
                <h4 className="font-accident-two-normal uppercase">{company}</h4>
                <h6 className="uppercase">{position}</h6>
                <p>{description}</p>
            </div>
        </div>
    </li>
)

ResumeJob.propTypes = { 
    orgin: PropTypes.string,
    durration: PropTypes.number,
    delay: PropTypes.number,
    distance: PropTypes.string,
    start: PropTypes.string.isRequired,
    end: PropTypes.string,
    company: PropTypes.string.isRequired,
    position: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
}

ResumeJob.defaultProps = {
    orgin: "left",
    durration: 400,
    delay: 400,
    distance: "25px",
    start: "",
    end: "",
    company: "",
    position: "",
    description: "",
}

export default ResumeJob;
