// maiku.net/components/ui/resume-school.jsx
import React from "react";
import PropTypes from "prop-types";

const ResumeSchool = ({ origin, duration, delay, distance, start, end, school, program, description }) => (
	<li>
		<div className="timeline-badge primary"><i className="flaticon-graduation61" /></div>
		<div
			className="timeline-panel"
			data-animation-origin={origin}
			data-animation-duration={duration}
			data-animation-delay={delay}
			data-animation-distance={distance}
		>
			<p className="timeline-time fontcolor-invert">{start} - {end}</p>
			<div className="timeline-photo timeline-mark-01" />
			<div className="timeline-heading">
				<h4 className="font-accident-two-normal uppercase">{school}</h4>
				<h6 className="uppercase">{program}</h6>
				<p>{description}</p>
			</div>
		</div>
	</li>
);

ResumeSchool.propTypes = {
	origin: PropTypes.string,
	duration: PropTypes.number,
	delay: PropTypes.number,
	distance: PropTypes.string,
	start: PropTypes.string.isRequired,
	end: PropTypes.string,
	school: PropTypes.string.isRequired,
	program: PropTypes.string.isRequired,
	description: PropTypes.string.isRequired,
};

ResumeSchool.defaultProps = {
	origin: "left",
	duration: 400,
	delay: 400,
	distance: "25px",
	start: "",
	end: "",
	school: "",
	program: "",
	description: "",
};

export default ResumeSchool;
