// maiku.net/components/ui/resume-timeline-project.jsx
import React from "react";
import PropTypes from "prop-types";

const TimeLineProject = ({origin, duration, delay, distance, project, projectRole, description, tech, screenshot}) => (
    <li>
        <div className="timeline-badge primary"><i className="flaticon-businessman-in-apresentation-with-a-graphic-on-a-board"></i></div>
        <div className="timeline-panel"
            data-animation-origin={origin}
            data-animation-duration={duration}
            data-animation-delay={delay}
            data-animation-distance={distance}>
            <div className="timeline-photo timeline-mark-01"></div>
            <div className="timeline-heading">
                <h4 className="font-accident-two-normal uppercase">{project}</h4>
                <h6 className="uppercase">{projectRole}</h6>
                <p>{description}</p>
                {tech && <h6 className="uppercase">{tech}</h6> }
            </div>
        </div>
    </li>
);

TimeLineProject.propTypes = { 
    orgin: PropTypes.string,
    durration: PropTypes.number,
    delay: PropTypes.number,
    distance: PropTypes.string,
    project: PropTypes.string.isRequired,
    projectRole: PropTypes.string.isRequired,
    tech: PropTypes.string,
    screenshot: PropTypes.string,
    description: PropTypes.string.isRequired,
}

TimeLineProject.defaultProps = {
    orgin: "left",
    durration: 400,
    delay: 400,
    distance: "25px",
    project: "",
    projectRole: "",
    tech: "",
    screenshot: "",
    description: "",
}

export default TimeLineProject;
