// maiku.net/components/pages/admin.jsx
import React, { Component } from "react";
import _ from "lodash";
//import { Route, Link } from 'react-router-dom'

import Layout from "maiku.net/components/ui/layout";
import SectionHeader from "maiku.net/components/ui/admin/section-header";
import PageTitle from "maiku.net/components/ui/page-title";
import Skills from "maiku.net/components/ui/admin/skills";
import About from "maiku.net/components/ui/admin/about";
import Info from "maiku.net/components/ui/admin/info";  
import Facts from "maiku.net/components/ui/admin/facts";
import Education from "maiku.net/components/ui/admin/education";

import base from "maiku.net/base";

export default class Admin extends Component {

    state = {
        section: "about",
        data: {
        }
    }

    componentWillMount() {
        this.ref = base.syncState(`maiku-net/data`,
            {
                context: this,
                state: "data",
                isNullable: true,
            });
    }

    componentWillUnmount() {
        base.removeBinding(this.ref);
    }

    addSkill = skill => {
        const skills = !this.state.data.skills
            ? [].concat(skill)
            : this.state.data.skills.concat(skill)

        this.setState({ data: { skills } })
    }

    removeSkill = skill => {
        const skills = this.state.data.skills.filter(item => item !== skill);
        this.setState({ data: { skills } })
    }

    updateSkill = update => {
        //-- find the original skill
        const skill = this.state.data.skills.find((item) => item._id === update._id);

        //-- update the original skill with the update values
        const result = Object.assign({}, { ...skill, ...update });

        //-- update the skill object
        const skills = this.state.data.skills.filter(item => item !== skill).concat(result);

        this.setState({ data: { skills } });
    }

    updateAboutMe = data => {
        this.setState({ data: { about: data } })
    }

    addInfo = information => {
        const info = !this.state.data.info
            ? [].concat(information)
            : this.state.data.info.concat(information)

        this.setState({ data: { info } })
    }

    updateInfo = update => {
        const information = this.state.data.info.find((item) => item._id === update._id);
        const result = Object.assign({},{ ...information, ...update}) 
        const info = this.state.data.info.filter(item => item !== information).concat(result);

        this.setState({ data: { info }});
    }

    removeInfo = information => {
        const info = this.state.data.info.filter(item => item !== information);
        this.setState({ data: { info } })
    }

    //-- facts
    addFact = fact => {
        const facts = !this.state.data.facts
            ? [].concat(fact)
            : this.state.data.facts.concat(fact)

        this.setState({ data: { facts } })
    }

    removeFact = fact => { 
        const facts = this.state.data.facts.filter(item => item !== fact);
        this.setState({ data: { facts }})
    }

    updateFact = update => {
        const fact = this.state.data.facts.find((item) => item._id === update._id);
        const result = Object.assign({},{ ...fact, ...update}) 
        const facts = this.state.data.facts.filter(item => item !== fact).concat(result);

        this.setState({ data: { facts }});
    }

    toggleSection = section => {
       this.setState({ section });
    }

    render() {
        return (
            <Layout>
                 <div className="content-wrap">

                    {/*//-- remove me later
                    <Link to="/admin/blah">go to blah</Link>*/}

                    <div id="feedback" className="inner-content">

                        <div className="container-fluid nopadding">

                            <div className="row nopadding">
                                <div className="col-md-3 nopadding"></div>
                                <div className="col-md-9 nopadding">

                                    <PageTitle
                                        name="Administration"
                                        origin="left"
                                        duration="600"
                                        delay="100"
                                        distance="50px"
                                    />

                                    <section className="inner-section feedback feedback-light bg-color01" data-section="feedback">

                                        <div className="padding-50">

                                            <div className="row">
                                                <div>
                                                    <div className="dividewhite2"></div>
                                                    <SectionHeader
                                                        toggled={this.state.section === "about"}
                                                        toggleSection={this.toggleSection}
                                                        name="about"
                                                    />
                                                    <SectionHeader
                                                        toggled={this.state.section === "info"}
                                                        toggleSection={this.toggleSection}
                                                        name="info"
                                                    />
                                                    <SectionHeader
                                                        toggled={this.state.section === "facts"}
                                                        toggleSection={this.toggleSection}
                                                        name="facts"
                                                    />
                                                    <SectionHeader
                                                        toggled={this.state.section === "employment"}
                                                        toggleSection={this.toggleSection}
                                                        name="employment"
                                                    />
                                                    <SectionHeader
                                                        toggled={this.state.section === "education"}
                                                        toggleSection={this.toggleSection}
                                                        name="education"
                                                    />
                                                    <SectionHeader
                                                        toggled={this.state.section === "skills"}
                                                        toggleSection={this.toggleSection}
                                                        name="skills"
                                                        isEnd={true}
                                                    />
                                                </div>

                                                {this.state.section === "about" ?
                                                    <About
                                                        data={this.state.data.about}
                                                        updateAboutMe={this.updateAboutMe}
                                                    />
                                                    : null
                                                }

                                                {this.state.section === "info" ?
                                                    <Info
                                                        data={this.state.data.info ? this.state.data.info : []}
                                                        addInfo={this.addInfo}
                                                        removeInfo={this.removeInfo}
                                                        updateInfo={this.updateInfo}
                                                    />
                                                    : null
                                                }

                                                {this.state.section === "skills" ?
                                                    <Skills
                                                        skills={this.state.data.skills ? this.state.data.skills : []}
                                                        addSkill={this.addSkill}
                                                        removeSkill={this.removeSkill}
                                                        updateSkill={this.updateSkill}
                                                    />
                                                    : null}

                                                {this.state.section === "facts" ? 
                                                <Facts
                                                    data={this.state.data.facts ? this.state.data.facts: []}
                                                    addFact={this.addFact}
                                                    removeItem={this.removeFact}
                                                    updateItem={this.updateFact}
                                                />
                                                    : null}  

                                                <Education
                                                    visibile={this.state.section === "education"}
                                                />

                                                <div className="col-md-12 divider-dynamic"></div>
                                            </div>

                                        </div>

                                        <div className="dividewhite6"></div>
                                    </section>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </Layout>
        );
    }
}
