// maiku.net/components/pages/admin.jsx
import React, { Component } from "react";
import _ from "lodash";

import Layout from "maiku.net/components/ui/layout";
import SectionHeader from "maiku.net/components/ui/admin/section-header";
import PageTitle from "maiku.net/components/ui/page-title";
import Skills from "maiku.net/components/ui/admin/skills";
import About from "maiku.net/components/ui/admin/about";

import base from "maiku.net/base";

export default class Admin extends Component {

    state = {
        section: "about",
        data: {
            about: "",
            description: [],
            facts: [],
            employment: [],
            education: {},
            skills: []
        }
    }

    componentWillMount() {
        this.ref = base.syncState(`maiku-net/data`,
            {
                context: this,
                state: "data",
                isNullable: true,
            });
    }

    componentWillUnmount() {
        base.removeBinding(this.ref);
    }

    addSkill = skill => {
        const skills = !this.state.data.skills
            ? [].concat(skill)
            : this.state.data.skills.concat(skill)

        this.setState({ data: { skills } })
    }

    removeSkill = skill => {
        const skills = this.state.data.skills.filter(item => item !== skill);
        this.setState({ data: { skills } })
    }

    updateSkill = update => {
        //-- find the original skill
        const skill = this.state.data.skills.find((item) => item._id === update._id);

        //-- update the original skill with the update values
        const result = Object.assign({}, { ...skill, ...update });

        //-- update the skill object
        const skills = this.state.data.skills.filter(item => item !== skill).concat(result);

        this.setState({ data: { skills } });
    }

    toggleSection = section => {
        this.setState({ section });
    }

    render() {
        return (
            <Layout>
                <div className="content-wrap">

                    <div id="feedback" className="inner-content">

                        <div className="container-fluid nopadding">

                            <div className="row nopadding">
                                <div className="col-md-3 nopadding"></div>
                                <div className="col-md-9 nopadding">

                                    <PageTitle
                                        name="Administration"
                                        origin="left"
                                        duration="600"
                                        delay="100"
                                        distance="50px"
                                    />

                                    <section className="inner-section feedback feedback-light bg-color01" data-section="feedback">

                                        <div className="padding-50">

                                            <div className="row">
                                                <div>
                                                    <div className="dividewhite2"></div>
                                                    {/*<a><h3
                                                        onClick={() => this.toggleSection("about")}
                                                        style={{ color: this.state.section !== "about" ? "#b7b7b7" : ""}}
                                                    className="font-accident-two-normal uppercase pull-left padding-0-15-0-0">About</h3></a>*/}
                                                    <SectionHeader
                                                        toggled={this.state.section === "about"}
                                                        toggleSection={this.toggleSection}
                                                        name="about"
                                                         />
                                                    <SectionHeader
                                                        toggled={this.state.section === "info"}
                                                        toggleSection={this.toggleSection}
                                                        name="info"
                                                         />
                                                    <a><h3
                                                        onClick={() => this.toggleSection("info")}
                                                        style={{ color: this.state.section !== "info" ? "#b7b7b7" : ""}}
                                                        className="font-accident-two-normal uppercase pull-left padding-0-15-0-0">Info</h3></a>
                                                    <a><h3
                                                        onClick={() => this.toggleSection("facts")}
                                                        style={{ color: this.state.section !== "facts" ? "#b7b7b7" : ""}}
                                                        className="font-accident-two-normal uppercase pull-left padding-0-15-0-0">Facts</h3></a>
                                                    <a><h3
                                                        onClick={() => this.toggleSection("employment")}
                                                        style={{ color: this.state.section !== "employment" ? "#b7b7b7" : ""}}
                                                        className="font-accident-two-normal uppercase pull-left padding-0-15-0-0">Employment</h3></a>
                                                    <a><h3
                                                        onClick={() => this.toggleSection("education")}
                                                        style={{ color: this.state.section !== "education" ? "#b7b7b7" : ""}}
                                                        className="font-accident-two-normal uppercase pull-left padding-0-15-0-0">Education</h3></a>
                                                    <a><h3
                                                        onClick={() => this.toggleSection("skills")}
                                                        style={{ color: this.state.section !== "skills" ? "#b7b7b7" : ""}}
                                                        className="font-accident-two-normal uppercase">Skills</h3></a>
                                                </div>

                                                <About
                                                    visibile={this.state.section === "about"}
                                                />

                                                <Skills
                                                    visibile={this.state.section === "skills"}
                                                    skills={this.state.data.skills ? this.state.data.skills : []}
                                                    addSkill={this.addSkill}
                                                    removeSkill={this.removeSkill}
                                                    updateSkill={this.updateSkill}
                                                />

                                                <div className="col-md-12 divider-dynamic"></div>
                                            </div>

                                        </div>

                                        <div className="dividewhite6"></div>

                                    </section>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </Layout>
        );
    }
}
