// maiku.net/components/pages/feedback.jsx
import React, { Component } from "react";

import Layout from "maiku.net/components/ui/layout";
import PageTitle from "maiku.net/components/ui/page-title";

import base from "maiku.net/base";

export default class Feedback extends Component {

    state = {
        data: {
            info: []
        },
    }

    componentWillMount() {
        this.ref = base.syncState(`maiku-net/data`,
            {
                context: this,
                state: "data",
                isNullable: true,
            });
    }

    componentWillUnmount() {
        base.removeBinding(this.ref);
    }

    sendFeedback = event => {
        event.preventDefault();

        var form = new FormData(document.getElementById("feedback-form"));
        fetch("http://www.maiku.net/email.php", {
            method: "POST",
            body: form
        })
        .then(response => {
            console.log(response);
            this.form.reset();
        })
        .catch(error => {
            console.log(error);
        });

        this.form.reset();
    }

    render() {
        const { info } = this.state.data;
        return (
            <Layout info={info}>
                <div className="content-wrap">

                    <div id="feedback" className="inner-content">

                        <div className="container-fluid nopadding">

                            <div className="row nopadding">
                                <div className="col-md-3 nopadding"></div>
                                <div className="col-md-9 nopadding">

                                    <PageTitle
                                        name="Feedback"
                                        origin="left"
                                        duration="600"
                                        delay="100"
                                        distance="50px"
                                    />

                                    <section className="inner-section feedback feedback-light bg-color01" data-section="feedback">

                                        <div className="padding-50">

                                            <div className="row">

                                                <div className=""
                                                    data-animation-origin="top"
                                                    data-animation-duration="500"
                                                    data-animation-delay="400"
                                                    data-animation-distance="25px">
                                                    <div className="dividewhite2"></div>
                                                    <h3 className="font-accident-two-normal uppercase text-center">Drop Us a Line</h3>
                                                    <h5 className="font-accident-two-normal uppercase text-center subtitle">Working hard and making the success</h5>
                                                </div>

                                                <div className="col-md-12 e-centered"
                                                    data-animation-origin="top"
                                                    data-animation-duration="500"
                                                    data-animation-delay="600"
                                                    data-animation-distance="25px">
                                                    <div className="dividewhite4"></div>
                                                    <div id="form-messages"></div>
                                                    <form 
                                                        className="wpcf7-form"
                                                        id="feedback-form"
                                                        onSubmit={(event) => this.sendFeedback(event)} 
                                                        ref={(input) => {this.form = input}} 
                                                    >
                                                        <div className="field">
                                                            <input 
                                                                type="text" 
                                                                id="name" 
                                                                name="name" 
                                                                placeholder="Name" 
                                                                required 
                                                                ref={(input) => { this.name = input }}
                                                            />
                                                        </div>

                                                        <div className="field">
                                                            <input 
                                                                type="email" 
                                                                id="email" 
                                                                name="email" 
                                                                placeholder="Email" 
                                                                required 
                                                                ref={(input) => { this.email = input }}
                                                            />
                                                        </div>

                                                        <div className="field">
                                                            <textarea 
                                                                id="message" 
                                                                name="message" 
                                                                placeholder="Message" 
                                                                rows="7" 
                                                                cols="30" 
                                                                required
                                                                ref={(input) => { this.message = input }}
                                                            ></textarea>
                                                        </div>

                                                        <div className="dividewhite2"></div>

                                                        <div className="field text-center">
                                                            <button type="submit" className="btn btn-lg btn-darker">Send Message</button>
                                                        </div>
                                                    </form>
                                                </div>
                                                <div className="col-md-12 divider-dynamic"></div>
                                            </div>

                                        </div>

                                        <div className="dividewhite6"></div>

                                    </section>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </Layout>
        );
    }
}
