// maiku.net/components/pages/home.jsx
import React, { Component } from "react";

import Layout from "maiku.net/components/ui/layout";
import EmploymentPosition from "maiku.net/components/ui/employment-position";
import InfoTile from "maiku.net/components/ui/info-tile";

import base from "maiku.net/base";

export default class Home extends Component {
	state = {
		data: {
			about: "",
			info: [],
			employment: [],
		},
	};

	componentWillMount() {
		this.ref = base.syncState("maiku-net/data", {
			context: this,
			state: "data",
			isNullable: true,
		});
	}

	componentWillUnmount() {
		base.removeBinding(this.ref);
	}

	render() {
		const { about, info } = this.state.data;
		const employment = this.state.data.employment.map(item => ({
			_id: item._id,
			current: item.current,
			start: item.start,
			end: item.end,
			employer: item.company,
			position: item.position,
		}));

		//-- I don't like this, I don't want the whole object, just the position
		const currentPosition = employment.find(item => item.current);

		return (
			<Layout info={info}>
				<div className="content-wrap">

					<section className="container-fluid" data-section="home">

						<div className="row">
							<div className="col-md-10 nopadding" />
							<div className="col-md-2 col-sm-12 nopadding">
								<div
									className="ui-block-color05 cv-pdf"
									data-animation-origin="top"
									data-animation-duration="300"
									data-animation-delay="1400"
									data-animation-distance="100px"
								>
									<a className="hvr-sweep-to-right">
										<i className="flaticon-download149" title="Download CV.pdf" />
									</a>
								</div>
							</div>
						</div>
						<div className="row flex-row">
							<div className="col-md-3 nopadding">
								<div
									className="padding-0-30-30-0 dot-block"
									data-animation-origin="top"
									data-animation-duration="300"
									data-animation-delay="1700"
									data-animation-distance="100px"
								>
									<h5 className="font-accident-two-normal uppercase">Employment</h5>
									<div className="dividewhite1" />
									{employment.map(item => (
										<EmploymentPosition
											key={item._id}
											start={item.start}
											end={item.end}
											employer={item.employer}
											position={item.position}
										/>
									))}
								</div>
							</div>
							<div id="personal-2" className="col-md-6 flex-column light nopadding">
								<div className="row flex-panel nopadding">

									<div className="col-md-12 col-sm-6 col-xs-6 nopadding">
										<div className="row flex-row nopadding">
											<InfoTile label="status" value="Open to offers" />
											<InfoTile label="Desired Salary" value="To be Discussed" />
										</div>
									</div>
									<div className="col-md-12 col-sm-6 col-xs-6 nopadding">
										<div className="row flex-row nopadding">
											<InfoTile label="Job Type" value="Full-Time, Contract" />
											<InfoTile label="Mobility" value="Willing to Relocate" />
										</div>
									</div>
									{/* current possition */}
									<div className="col-md-12 col-sm-6 col-xs-6 nopadding">
										<div className="row flex-row nopadding">
											{currentPosition
												? <InfoTile label="Current Position" value={currentPosition.position} />
												: ""}
										</div>
									</div>
								</div>

							</div>
							<div className="col-md-1" />
							<div className="col-md-2 nopadding" />
						</div>

						{about !== ""
							? <div className="row nopadding">
									<div className="col-md-6 nopadding" />
									<div
										className="col-md-6 nopadding dot-block"
										data-animation-origin="right"
										data-animation-duration="500"
										data-animation-delay="2000"
										data-animation-distance="200px"
									>
										<h5 className="font-accident-two-normal uppercase">Overview</h5>
										<p>{about}</p>
									</div>
								</div>
							: ""}
					</section>

				</div>
			</Layout>
		);
	}
}
